libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java

```diff
@@ -120,15 +121,27 @@ public void fetchRouteFromOffRouteEvent(OffRouteEvent event) {
     }
   }
 
+  /**
+   * Sets the callback for reroute events and faster routes
+   *
+   * @param callback to set
+   */
+  public void setCallback(Callback callback) {
+    this.callback = callback;
+  }
+
   private void processRoute(@NonNull Response<DirectionsResponse> response) {
     if (isValidRoute(response)) {
       List<DirectionsRoute> routes = response.body().routes();
       DirectionsRoute bestRoute = routes.get(FIRST_ROUTE);
-      DirectionsRoute chosenRoute = route.getValue();
+      DirectionsRoute chosenRoute = route;
       if (isNavigationRunning(chosenRoute)) {
```

Comment:
We could also [decompose these conditionals](https://sourcemaking.com/refactoring/decompose-conditional), you know, we should keep `public` methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.).